content/reference/sql/sql-support/miscellaneous.md

```diff
@@ -113,6 +111,74 @@ select `index`, `position`, row_count, distinct_count, columns, upper_bound, upp
+
+Setting a non-zero threshold defers updates until after a certain fraction of chunks are edited. For example, a 100% difference threshold updates stats when:
+
+1) The table was previously empty and now contains data.
+
+2) The table grew or shrank such that the tree height grew or shrank, and therefore the target fanout level changed.
+
+3) Inserts added twice as many chunks.
+
+4) Deletes removed 100% of the preexisting chunks.
+
+5) 50% of the chunks were edited (an in-place edit deletes one chunk and adds one chunk, for a total of two changes relative to the original chunk)
+
+Any combination of edits/inserts/deletes that exceeds the trigger threshold will also update stats.
+
+We enable refresh with one mandatory and two optional system variables:
+
+```sql
+dolt sql -q "set @@PERSIST.dolt_stats_auto_refresh_enabled   = 1;"
+dolt sql -q "set @@PERSIST.dolt_stats_auto_refresh_interval  = 120;"
+dolt sql -q "set @@PERSIST.dolt_stats_auto_refresh_threshold = 0.5"
+```
+
+The first enables auto-refresh. It is a global variable that must be set during `dolt sql-server` startup and affects all databases in a server context. Databases added or dropped to a running server automatically opt-in to statistics refresh if enabled.
+
+The second two variables configure 1) how often a timer wakes up to check stats freshness (seconds), and 2) the threshold updating a table's active statistics (new+deleted/previous chunks as a percentage between 0-1). For example, `dolt_stats_auto_refresh_interval = 600` means the server only attempt to update stats every 10 minutes, regardless of how much a table has changed. Setting `dolt_stats_auto_refresh_threshold = 0` forces stats to update in response to any table change.
+
+A last variable blocks statistics from loading from disk on startup, or writing to disk on ANALYZE:
+
+```sql
+dolt sql -q "set @@PERSIST.dolt_stats_memory_only = 1"
+```
+
+### Performance
+
+Lowering check intervals and update thresholds increases the refresh read and write load. Refreshing statistics uses shortcuts to avoid reading from disk when possible, but in most cases at least needs to read the target fanout level of the tree from disk to compare previous and current chunk sets. Exceeding the refresh threshold reads all data from disk associated with the new chunk ranges, which will be the most expensive impact of auto-refresh. Dolt uses ordinal offsets to avoid reading unnecessary data, but the tree growing or shrinking by a level forces a full tablescan.
+
+For example, setting the check interval to 0 seconds (constant), the update threshold to 0 (any change triggers refresh) reduces the `oltp_read_write` sysbench benchmark's throughput by 15%. An increase in the update threshold for a 0-interval reduces throughput even more. On the other hand, basically any non-zero interval reduces the fraction of time spent performing stats updates to a negligible level:
```

Comment:
What does increasing the threshold reduce throughput? Surely batching the update is more efficient than running it every time the tables change, right?

Reply:
I'm not sure yet, surprising but it appeared reproducible. I'll rerun benchmarks after adding the controller api